### ISSUE TITLE
Sample code in CUDA EP does not compile

### ISSUE THREAD

**Issue by natke** [microsoft/onnxruntime#11322]
**Describe the bug**
This code does not compile: https://onnxruntime.ai/docs/execution-providers/CUDA-ExecutionProvider.html#cc

```
/home/azureuser/cloudfiles/code/Users/nakersha/onnxruntime-inference-examples/c_cxx/api/main.cpp:32:58: error: ‘EXHAUSTIVE’ is not a member of ‘OrtCudnnConvAlgoSearch’
 options.cudnn_conv_algo_search = OrtCudnnConvAlgoSearch::EXHAUSTIVE;
                                                          ^~~~~~~~~~
/home/azureuser/cloudfiles/code/Users/nakersha/onnxruntime-inference-examples/c_cxx/api/main.cpp:35:1: error: ‘SessionOptionsAppendExecutionProvider_CUDA’ was not declared in this scope
 SessionOptionsAppendExecutionProvider_CUDA(session_options, &options);
```

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- ONNX Runtime installed from (source or binary): binary
- ONNX Runtime version: 1.11
- Python version: N/A
- Visual Studio version (if applicable): N/A
- GCC/Compiler version (if compiling from source): 7.5
- CUDA/cuDNN version: 11.4
- GPU model and memory: V100

**To Reproduce**
- Add code to a C++ application

**Expected behavior**
Code should compile

**Additional context**
Add any other context about the problem here. If the issue is about a particular model, please share the model details as well to facilitate debugging.

**Comment by RyanUnderhill** [comment]
Hmm, first thought is it should be `OrtCudnnConvAlgoSearchExhaustive` as that's how it's defined in the C API: https://github.com/microsoft/onnxruntime/blob/95a16c1ffe93c061c70e2d4ae4b0950bf40ae2d0/include/onnxruntime/core/session/onnxruntime_c_api.h#L352

**Comment by natke** [comment]
Thank you @RyanUnderhill!

**Comment by natke** [comment]
Do you think we would also need to fix this line?

https://onnxruntime.ai/docs/execution-providers/CUDA-ExecutionProvider.html#cudnn_conv_algo_search

**Comment by natke** [comment]
Fixed by #11777
